libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -64,12 +64,30 @@
     #error "IOT_MQTT_RETRY_MS_CEILING cannot be 0 or negative."
 #endif
 
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Each compilation unit must define the NetworkContext struct.
+ */
+struct NetworkContext
+{
+    MqttTransportParams_t * pParams;
+};
+
+/*-----------------------------------------------------------*/
+
 /**
  * @brief Fixed Size Array to hold Mapping of MQTT Connection used in MQTT 201906.00 library to MQTT Context
  * used in calling MQTT LTS API from shim to send packets on the network.
  */
 _connContext_t connToContext[ MAX_NO_OF_MQTT_CONNECTIONS ];
 
+/**
+ * @brief Represents the network context used for the TLS session with the
+ * server.
+ */
+static NetworkContext_t networkContext = { 0 };
```

Comment:
This is static now. Could this cause problems? Can we possibly have a race condition with `pParams` when two tasks simultaneously make a connection?

Reply:
Thanks for the suggestion, fixed ini c1b87014da86c6e5d22bc60ad0efe824c5ef873c